components/feature/addons/src/main/java/mozilla/components/feature/addons/amo/AddonCollectionProvider.kt

```diff
@@ -84,14 +86,21 @@ class AddonCollectionProvider(
      * @param readTimeoutInSeconds optional timeout in seconds to use when fetching
      * available add-ons from a remote endpoint. If not specified [DEFAULT_READ_TIMEOUT_IN_SECONDS]
      * will be used.
+     * @param language indicates in which language the translatable fields should be in, if no
+     * matching language is found then a fallback translation is returned using the default language.
+     * When it is null all translations available will be returned.
      * @throws IOException if the request failed, or could not be executed due to cancellation,
      * a connectivity problem or a timeout.
      */
     @Throws(IOException::class)
     @Suppress("NestedBlockDepth")
-    override suspend fun getAvailableAddons(allowCache: Boolean, readTimeoutInSeconds: Long?): List<Addon> {
-        val cachedAvailableAddons = if (allowCache && !cacheExpired(context)) {
-            readFromDiskCache()
+    override suspend fun getAvailableAddons(
+        allowCache: Boolean,
+        readTimeoutInSeconds: Long?,
+        language: String?
+    ): List<Addon> {
+        val cachedAvailableAddons = if (allowCache && !cacheExpired(context, language, useFallbackFile = false)) {
+            readFromDiskCache(language, useFallbackFile = false)
```

Comment:
@csadilek I found a bug while testing. For example, if I started with `en` as  language and I switch to `es` we expected that if I have internet connection that json for `es` will be shown (add-ons show in Spanish), but that wasn't happening, we were always using a fallback file behavior that was causing the cache to not be expired. For this reason, I introduced the `useFallbackFile` parameter to be exact when a fallback file should be used.